hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieAppendHandle.java

```diff
@@ -652,6 +660,16 @@ private static Map<HeaderMetadataType, String> getUpdatedHeader(Map<HeaderMetada
     if (addBlockIdentifier && !HoodieTableMetadata.isMetadataTable(config.getBasePath())) { // add block sequence numbers only for data table.
       updatedHeader.put(HeaderMetadataType.BLOCK_IDENTIFIER, attemptNumber + "," + blockSequenceNumber);
     }
+    if (config.shouldWritePartialUpdates()) {
+      // When enabling writing partial updates to the data blocks, the full schema is also written
+      // to the block header so that the reader can differentiate partial updates vs schema
+      // evolution, based on the "SCHEMA" which contains the partial schema and the "FULL_SCHEMA"
+      // which contains the full schema of the table at this time.
+      updatedHeader.put(
+          HeaderMetadataType.FULL_SCHEMA,
+          HoodieAvroUtils.addMetadataFields(
+              getWriteSchema(config), config.allowOperationMetadataField()).toString());
```

Comment:
`getWriteSchema(config)` fetches the full schema at this particular commit.  For schema evolution, I think it's better to keep the full schema in each log block, as the schema can evolve across log blocks.

Reply:
I'm fine if we deem the partial update as a sepecial use case for schema evolution, then the full schema should always be the current latest table schema, we should not encode it in the log block because it is evolved automically.

Reply:
I agree that the partial update is a sepecial use case for schema evolution and the full schema should always be the current latest table schema or the full schema of a particular commit time for time travel query.